### PR TITLE
[Resource] Fix internal server error when apply sorting or filtering

### PR DESCRIPTION
**Pull request by Arminek**
| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | no |
| BC breaks?      | no |
| License         | MIT |

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
Shouldn't we check this inside of the repository? To avoid this error also when someone uses it via php code, not yml (code, hehe)?

File: src/Sylius/Bundle/ResourceBundle/Controller/ResourcesResolver.php [hunk 1]
```diff
@@ -46,4 +60,42 @@ public function getResources(RequestConfiguration $requestConfiguration, Reposit
...
         return $repository->findBy($criteria, $sorting, $requestConfiguration->getLimit());
     }
+
+    /**
+     * @param array $sorting
+     * @param RepositoryInterface $repository
+     *
+     * @return array
+     */
+    private function unsetUnexistSortingFields(array $sorting, RepositoryInterface $repository)
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ResourceBundle/Controller/ResourcesResolver.php]: The only difference between these methods is first argument name... We can merge them

**Comment by pamil** [comment]
This will crash sorting / filtering on properties of joined objects (sort by price on products list on taxon page).

**Comment by bendavies** [comment]
This is wrong IMO. This will break where users have implemented custom criteria filtering inside their repository inside `applyCriteria`. A use might have a custom criteria name that is not a field name on the entity. So, i agree with @pjedrzejewski, this should be done inside the repository.

**Comment by Arminek** [comment]
This yaml coding style makes me sick, of course you are right 💃 .

**Comment by Arminek** [comment]
This solution wont work with joined properties.

**Comment by pjedrzejewski** [comment]
Nice work Arek! Regarding joined fields, ``indexAction`` is generic and should be used wisely. For proper sorting and filtering we recommend using grids.

**Pull request merged**